runtime/queries/metricsview.go

```diff
@@ -160,7 +160,12 @@ func buildFilterClauseForCondition(cond *runtimev1.MetricsViewFilter_Cond, exclu
 		questionMarks := strings.Join(repeatString("?", len(cond.In)-nullCount), ",")
 		// <dimension> (NOT) IN (?,?,...)
 		if questionMarks != "" {
-			clauses = append(clauses, fmt.Sprintf("%s %s IN (%s)", cond.Name, operatorPrefix, questionMarks))
+			clause := fmt.Sprintf("%s %s IN (%s)", cond.Name, operatorPrefix, questionMarks)
+			if nullCount == 0 && exclude {
+				// In case of exclusion, we need to explicitly include NULL value
+				clause = fmt.Sprintf("%s OR (%s IS NULL)", clause, cond.Name)
+			}
```

Comment:
Instead of duplicating `OR (%s IS NULL)` across every `IN` and `ILIKE` term, can't we just add it as a single `OR %s IS NULL` to the `clause` at the end of the function?